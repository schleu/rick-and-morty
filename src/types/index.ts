  
  export const status = ["Alive", "Dead", "unknown"] as const
  export type Status = typeof status[number];
  
  export const gender = ["Female", "Male", "Genderless",  "unknown"] as const
  export type Gender = typeof gender[number]
  
  export interface iCharacter  {
    "id": number,
    "name": string,
    "status": Status,
    "species": string,
    "type": string,
    "gender": Gender,
    "origin": {
        "name": string,
        "url": string,
    },
    "location": {
        "name": string,
        "url": string,
    },
    "image": string;
    "episode": string[],
    "url": string,
    "created": string,
  }

  export interface iFilterCharacter {
    name?:string;
    status?:Status;
    gender?:Gender;
    species?:string;
    type?:string;
  }

  interface iFilterEpisode{
    name:string;
    code: number;
  }


export  interface iFilter{
    page?:number;
    character?:iFilterCharacter;
    epsode?:iFilterEpisode
  }

export interface response {
    info:{
      "count": number,
      "pages": number,
      "next": string,
      "prev": string
  },
    results:iCharacter[]
  }



  export interface iEpisode{
    id:number;
    name:string;
    air_date:string;
    episode: string;
    characters: string[];
    url: string;
    created:string;
  }