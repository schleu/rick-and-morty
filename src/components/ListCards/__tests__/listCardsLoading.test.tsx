import { iCharacter } from '@/types';
import { render, screen } from '@testing-library/react';
import { ListCard } from '../Xadwa_index';

import '@testing-library/jest-dom'

const mockCharacters: iCharacter[] = [
    {
        "id": 1,
        "name": "Rick Sanchez",
        "status": "Alive",
        "species": "Human",
        "type": "",
        "gender": "Male",
        "origin": {
            "name": "Earth (C-137)",
            "url": "https://rickandmortyapi.com/api/location/1"
        },
        "location": {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3"
        },
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "episode": [
            "https://rickandmortyapi.com/api/episode/1",
            "https://rickandmortyapi.com/api/episode/2",
            "https://rickandmortyapi.com/api/episode/3",
        ],
        "url": "https://rickandmortyapi.com/api/character/1",
        "created": "2017-11-04T18:48:46.250Z"
      },
      {
        "id": 2,
        "name": "Morty Smith",
        "status": "Alive",
        "species": "Human",
        "type": "",
        "gender": "Male",
        "origin": {
            "name": "unknown",
            "url": ""
        },
        "location": {
            "name": "Citadel of Ricks",
            "url": "https://rickandmortyapi.com/api/location/3"
        },
        "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        "episode": [
            "https://rickandmortyapi.com/api/episode/1",
        ],
        "url": "https://rickandmortyapi.com/api/character/2",
        "created": "2017-11-04T18:50:21.651Z"
    },
    {
        "id": 3,
        "name": "Summer Smith",
        "status": "Alive",
        "species": "Human",
        "type": "",
        "gender": "Female",
        "origin": {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20"
        },
        "location": {
            "name": "Earth (Replacement Dimension)",
            "url": "https://rickandmortyapi.com/api/location/20"
        },
        "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
        "episode": [
            "https://rickandmortyapi.com/api/episode/6",
            "https://rickandmortyapi.com/api/episode/7"
        ],
        "url": "https://rickandmortyapi.com/api/character/3",
        "created": "2017-11-04T19:09:56.428Z"
    }
];

describe('<ListCard />', () => {
    it('screenshot loading',()=>{
        const {container} = render(<ListCard characters={mockCharacters} isLoading={true} />);
        expect(container).toMatchSnapshot()
    })

    it('renders loading skeletons when isLoading is true', () => {
        render(<ListCard characters={mockCharacters} isLoading={true}/>)
        const skeletons = screen.getAllByTestId('card-skeleton');
        expect(skeletons.length).toBe(20);
    });
});
