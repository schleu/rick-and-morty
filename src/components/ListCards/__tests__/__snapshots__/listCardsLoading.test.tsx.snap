// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCard /> screenshot loading 1`] = `
<div>
  <div
    class="description"
  >
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
    <div
      class="card"
      data-testid="card-skeleton"
    >
      <div
        class="cardSkeletonImage skeleton"
      />
      <div
        class="cardDescription"
      >
        <div
          class="cardTitle"
        >
          <h3
            class="skeleton cardSkeletonTitle"
          />
          <div
            class="cardSkeletonFavorite skeleton"
          />
        </div>
        <p
          class="cardSkeletonStatus skeleton"
        />
        <p
          class="characterStatus skeleton"
        >
          <span
            class="characterStatusPointer skeleton"
          />
        </p>
      </div>
    </div>
  </div>
</div>
`;
